src/simplewallet/simplewallet.cpp

```diff
@@ -6641,40 +6649,53 @@ bool simple_wallet::lns_print_owners_to_names(const std::vector<std::string>& ar
         fail_msg_writer() << "arg contains non-hex characters: " << arg;
         return false;
       }
-      request.entries.push_back(arg);
+
+      if (request->entries.size() >= cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::MAX_REQUEST_ENTRIES)
+      {
+        requests.emplace_back();
+        request = &requests.back();
+      }
+      request->entries.push_back(arg);
     }
+
   }
 
-  std::vector<cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::response_entry> entries = m_wallet->lns_owners_to_names(request, failed);
-  if (failed)
+  rpc_results.reserve(requests.size());
+  for (auto const &request : requests)
   {
-    fail_msg_writer() << *failed;
-    return false;
+    std::vector<cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::response_entry> result = m_wallet->lns_owners_to_names(request, failed);
+    if (failed)
+    {
+      fail_msg_writer() << *failed;
+      return false;
+    }
+    rpc_results.emplace_back(std::move(result));
   }
 
-  for (cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::response_entry const &entry : entries)
+  for (size_t i = 0; i < rpc_results.size(); i++)
   {
-    std::string const *owner = &my_key;
-    if (args.size()) // If specified owner to look up, retrieve it
+    std::vector<cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::response_entry> const &rpc = rpc_results[i];
+    for (cryptonote::COMMAND_RPC_LNS_OWNERS_TO_NAMES::response_entry const &entry : rpc)
```

Comment:
This seems really verbose compared to:
```C++
for (const auto& rpc : rpc_results)
  for (const auto& entry : rpc)
  {
    // ...
  }
```

Reply:
Addressed. But kept the index i for indexing into the request